documents/invoice/service.go

```diff
@@ -122,7 +122,7 @@ func (s service) calculateDataRoot(ctx context.Context, old, new documents.Model
 }
 
 // Create takes and invoice model and does required validation checks, tries to persist to DB
-func (s service) Create(ctx context.Context, inv documents.Model) (documents.Model, uuid.UUID, error) {
+func (s service) Create(ctx context.Context, inv documents.Model, txID uuid.UUID) (documents.Model, uuid.UUID, error) {
```

Comment:
This is a great example of a parameter that needs to be available down the stream. I would add it as part of the `ctx`

Reply:
I agree!